common/djangoapps/xblock_django/user_service.py

```diff
@@ -24,10 +29,18 @@ class DjangoXBlockUserService(UserService):
     A user service that converts Django users to XBlockUser
     """
     def __init__(self, django_user, **kwargs):
+        """
+        Constructs a DjangoXBlockUserService object.
+
+        Args:
+            user_is_staff(bool): optional flag to mark staff users.
+            anonymous_user_id(str): optional string to override the default anonymous_user_id
```

Comment:
Can you clarify what you mean by `override`? Does `django_user.anonymous_user_id` already have a value?

Reply:
See https://github.com/edx/edx-platform/pull/28440/commits/ad0a247ab52363efc0281ff2df565b53d0c3b000#diff-4e10e4b8dffa69869f2098d333e9dc71c7e56b82cd08529bec0cc337aab9b724R37

Reply:
I think these would be more useful:

```
user_is_staff(bool): optional - whether the user is staff in the course
anonymous_user_id(str): optional - anonymous_user_id for the user in the course
```

Reply:
Addressed by 6ae6f6b